src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,49 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int LOGIN_MIN_LENGTH = 6;
+    private static final int PASSWORD_MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new UserRegistrationException("Can't register user, because user is null");
+        }
+        if (user.getAge() == null) {
+            throw new UserRegistrationException("The age of the user can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new UserRegistrationException("The password can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new UserRegistrationException("The login can't be null");
+        }
+        if (user.getLogin().length() < LOGIN_MIN_LENGTH) {
+            throw new UserRegistrationException("The length of the login must be at least "
+                    + LOGIN_MIN_LENGTH);
+        }
+        if (user.getPassword().length() < PASSWORD_MIN_LENGTH) {
+            throw new UserRegistrationException("The length of the password must be at least "
+                    + PASSWORD_MIN_LENGTH);
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new UserRegistrationException("The age of the user must be at least " + MIN_AGE);
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new UserRegistrationException("User already exists with login "
+                    + user.getLogin());
```

Comment:
maybe it is better to have this checks in separated methods? and then call them one by one